common/stats/status_server.h

```diff
@@ -35,15 +38,32 @@ class Stats;
 
 class StatusServer {
  public:
+  /*
+   * \brief Endpoint/function name to std::function mapping.
+   *
+   * Stores the mapping of an end-point/function to the the
+   * actual std::function to be executed when requested.
+   *
+   * The std::function takes as input the list of arguments to
+   * the function in the form of a vector of [key, value] pairs
```

Comment:
The way it is now is to return a dynamic, which is always converted to json unless it ends with .txt. The benefit of this that we avoid repeating the call to serialize to json in all handlers.